### ISSUE TITLE
Export translations as .ts file to enable static type safety

### ISSUE THREAD

**Issue by samuelstroschein** [samuelstroschein/inlang#13]
One of the big problems of i18n is no existent type safety. [https://github.com/ivanhofer/typesafe-i18n](https://github.com/ivanhofer/typesafe-i18n) is gaining momentum as one of the first solutions that enable typesafety for i18n. It requires setup and continuously watches the translation file to auto-generate type annotations. 

Inlang currently posts missing translations via a network request. That is already better than most wide-stream i18n solutions but also has downsides: 

1. During development, the missing translations can become convoluted. A developer might wrote a typo "hrllo" instead of "hello" which the dev fixes after 1 min but it's too late "hrllo" and "hello" are posted and tracked as missing translations. One way to avoid that is using ids `t('hrrlo', {id: 'compenent.hello'} ` . 
2. Even if id's are used, the translation for 'component.hello' might not exist yet. In other words the DX can be improved by showing a linting error that the translation key does not exist yet. And instead of baking a network request into the t function, a VSCode extension could do that job. I currently imagine the linting error, when hovering over the error a small popup from the inlang-vscode extension shows with a simple "add to translations" button. 

Regardless of how missing translations are tracked. Typesafety is even better for variables `` t("Hello {user}", vars: { uset: "Samuel" }) `` is a runtime error. The type `` uset `` is not statically caught. Neither would be something like `` t("Hello {userId:number}", vars: {userId: "Samuel"}` `` userId is expected to be a `` number `` but a string is passed. As a matter of fact, that will not even be a runtime error. 

### TLDR

Exporting statically typed translation files into the source repository instead of json files can greatly improve DX.

___

Automatically created from [Linear (#25)](https://linear.app/inlang/issue/INL-25/export-translations-as-ts-file-to-enable-static-type-safety)

*via LinearSync*
